### PR TITLE
User profile page

### PR DESCRIPTION
**Pull request by zeus999**

### CODE REVIEW

File: src/app/reducers/user-trips.reducer.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export function reducer(state = initialState, action: Action): State {
 		case ActionTypes.LOAD_USER_TRIPS_SUCCESS: {
 			const Trips = action.payload;
 			const userId = state.selectedUserId;
-			const	newTrips = Trips.filter(trip => !state.userTrips[userId][trip.id]);
+			const	newTrips = Trips.filter(trip => !state.userTrips[userId].trips[trip.id]);
```
Comment (ashish173) [inline src/app/reducers/user-trips.reducer.ts]: extra space here

File: src/app/components/user/user.routes.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
+import { UserProfileRoutes } from './user-profile/user-profile.routes';
+import { UserProfileComponent } from './user-profile/user-profile.component';
 import { UserSettingsComponent } from './user-settings/user-settings.component';
 
 export const UserRoutes = [
-	{ path: 'settings', component: UserSettingsComponent }
+	{ path: 'settings', component: UserSettingsComponent },
+	{ path: ':id', component: UserProfileComponent, children: UserProfileRoutes }
+	// { path: ':id', component: UserProfileComponent }
```
Comment (ashish173) [inline src/app/components/user/user.routes.ts]: remove this commented code.

File: src/app/components/user/user-profile/user-trips/user-trips.component.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  styleUrls: ['./user-trips.component.scss']
+})
+export class UserTripsComponent implements OnInit {
+  userTrips$: Observable<Trip[]>;
+
+  constructor(private store: Store<fromRoot.State>) {
+    this.userTrips$ = this.store.select(fromRoot.getUserTripsCollection);
+   }
+
+  ngOnInit() {
```
Comment (ashish173) [inline src/app/components/user/user-profile/user-trips/user-trips.component.ts]: remove this for now .

File: src/app/components/user/user-profile/user-trips/user-trips.component.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  selector: 'tr-user-trips',
+  templateUrl: './user-trips.component.html',
+  styleUrls: ['./user-trips.component.scss']
+})
+export class UserTripsComponent implements OnInit {
+  userTrips$: Observable<Trip[]>;
+
+  constructor(private store: Store<fromRoot.State>) {
+    this.userTrips$ = this.store.select(fromRoot.getUserTripsCollection);
+   }
```
Comment (ashish173) [inline src/app/components/user/user-profile/user-trips/user-trips.component.ts]: indentation not correct.

File: src/app/components/user/user-profile/user-profile.component.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+})
+export class UserProfileComponent implements OnInit {
+
+  private subscription: Subscription;
+  private userIndex: string;
+
+  constructor(private store: Store<State>, private activatedRoute: ActivatedRoute) {}
+
+  ngOnInit() {
+    this.subscription = this.activatedRoute.params.subscribe(
```
Comment (pkrawat1) [inline src/app/components/user/user-profile/user-profile.component.ts]: unsubscribe it

**Pull request merged**